### PR TITLE
PT-3278: Export doesn't take into account the filters for genes and f…

### PR DESCRIPTION
**Pull request by marta-**
…or clinical diagnosis

Refactored the ExportFilter service to reuse the generic query builder of the LiveTable instead of generating the query from scratch filter by filter.

### CODE REVIEW

**Comment by sashaandjic** [comment]
Push and Export dialogs doesn't take into account the user who's logged when filtering. E.g. in the following xar https://drive.google.com/open?id=0B1t2Eth-aBrha3JkRG5FcG9SbG8 I have user JohnDoe (password 123456) who has 4 patients with BRCA1 gene. When I log as MaryGreen (password 123456) who has no patients with BRCA1, export and push dialog show 4 at the bottom (John's patients). If I proceed with push/export nothing gets pushed/exported. Push fails, export file is empty. Advanced filtering option works as expected.

**Comment by marta-** [comment]
@sashaandjic The problem you reported should be fixed by the last commit, please retest.

**Comment by sashaandjic** [comment]
Tested. No problems found.

File: components/navigation/ui/src/main/resources/PhenoTips/LiveTableMacros.xml [hunk 1]
```diff
@@ -0,0 +1,672 @@
...
+ #end
+#end
+
+#**
+ * Adds information about each live table row to the JSON returned by a live table data call.
+ * NOTE: This macro uses Velocity variables defined by gridresultwithfilter_buildQuery.
+ *
+ * @param $map stores the JSON in memory so that it can be adjusted before serialization
+ *#
+#macro(gridresult_buildRowsJSON $map)
```
Comment (sdumitriu) [inline components/navigation/ui/src/main/resources/PhenoTips/LiveTableMacros.xml]: This is also defined identically at line 152, why the duplicate?

File: components/export/ui/src/main/resources/PhenoTips/ExportFilter.xml [hunk 2]
```diff
@@ -37,231 +37,34 @@
...
-      #set ($query = ", DBStringListProperty disease ${query} and disease.id.id = obj.id and disease.id.name = 'omim_id' and (? in elements(disease.list)")
-      #set ($isFirstDisease = false)
-    #else
-      #set ($query = $query.concat(" ${joinOperator} ? in elements(disease.list)"))
-    #end
-    #set ($discard = $parameters.add($disease))
-  #end
+#set ($extra = " $!{request.filterWhere} ")
+#set ($params = [])
+#if ("$!request.space" != '')
```
Comment (sdumitriu) [inline components/export/ui/src/main/resources/PhenoTips/ExportFilter.xml]: I wonder where these came from...

File: components/patient-data-sharing/push-ui/src/main/resources/PhenoTips/PushPatient.xml [hunk 3]
```diff
@@ -1224,7 +1224,7 @@
                     onSuccess: function(response) {
                         var patients = [];
                         response.responseJSON.each(function(item) {
-                            patients.push(item.replace("data.",""));
+                            patients.push(item.substring(item.lastIndexOf('.') + 1));
```
Comment (sdumitriu) [inline components/patient-data-sharing/push-ui/src/main/resources/PhenoTips/PushPatient.xml]: Would it be safer to check first if `contains('.')`?

Comment (marta-) [inline components/patient-data-sharing/push-ui/src/main/resources/PhenoTips/PushPatient.xml]: Unnecessary. `lastIndexOf` returns -1 if `.` is not there, and then `substring` just returns the whole string.

File: components/navigation/ui/src/main/resources/PhenoTips/LiveTableMacros.xml (hunk 1 above)
Comment (marta-) [inline components/navigation/ui/src/main/resources/PhenoTips/LiveTableMacros.xml]: Leftover copy/paste, thanks for spotting it.

File: components/export/ui/src/main/resources/PhenoTips/ExportFilter.xml (hunk 2 above)
Comment (marta-) [inline components/export/ui/src/main/resources/PhenoTips/ExportFilter.xml]: From nowhere... that's why the filtering didn't work properly in Push and Export for non-admins. Sasha reported the issue and it was fixed in 67b4373 .

**Comment by marta-** [comment]
@sdumitriu your comments were addressed.

**Pull request merged**